contracts/finality/src/exec/finality.rs

```diff
@@ -347,16 +338,15 @@ fn slash_finality_provider(
         )
         .map_err(|err| ContractError::SecretKeyExtractionError(err.to_string()))?;
 
-    // Emit slashing event.
-    // Raises slashing event to babylon over IBC.
-    let msg = ExecuteMsg::Slashing {
-        sender: info.sender.clone(),
-        evidence: evidence.clone(),
-    };
-    let wasm_msg: WasmMsg = WasmMsg::Execute {
-        contract_addr: env.contract.address.to_string(),
-        msg: to_json_binary(&msg)?,
-        funds: vec![],
+    let msg = BabylonMsg::EquivocationEvidence {
+        signer: info.sender.to_string(),
```

Comment:
Not sure if this is the right signer for the equivocation evidence message.

Reply:
Babylon side does not validate the signer because anyone could submit an equivocation evidence, as long as the evidence is verified. So this does not matter